source/RoslynAnalyzers/Descriptors.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace Octopus.RoslynAnalyzers
+{
+    public static class Descriptors
+    {
+        static DiagnosticDescriptor GetTestAnalyzerDescriptor(string id, string title, string? description = null)
+        {
+            return new DiagnosticDescriptor(id,
+                title,
+                title,
+                "Octopus.Testing",
+                DiagnosticSeverity.Warning,
+                false, // Testing analysers are disabled by default, these will only be enabled for test projects
+                description);
+        }
+
+        static readonly string noBaseClassesDescription = "Creating base classes for tests is a really good way to make tests neat, and 1 or 2 levels " +
```

Comment:
Pulled a lot of these descriptions from the convention tests that we'll be replacing with these analysers.

Reply:
👍